packages/core/content-manager/server/src/controllers/single-types.ts

```diff
@@ -281,19 +280,21 @@ export default {
     }
 
     ctx.body = await pipeAsync(
-      (document) => entityManager.discard(document, model, { locale }),
+      () => singleTypes.discard(model, { locale }),
       permissionChecker.sanitizeOutput,
       (document) => documentMetadata.formatDocumentWithMetadata(model, document)
-    )(document);
+    )();
   },
 
   async countDraftRelations(ctx: any) {
     const { userAbility } = ctx.state;
     const { model } = ctx.params;
-
-    const entityManager = getService('entity-manager');
+    const { query } = ctx.request;
+    const singleTypes = getService('single-types');
     const permissionChecker = getService('permission-checker').create({ userAbility, model });
 
+    const { locale } = getDocumentDimensions(query);
```

Comment:
Not sure we want that generic dimensions term as of now :)

Reply:
sure! let me change it